run_xnat2bids.py

```diff
@@ -83,83 +135,90 @@ def main():
 
-        # Update default config with user provided parameters
-        default_slurm.update(user_slurm)
-        default_x2b.update(user_x2b)
-
-        # Assemble final argument list 
-        merged_dict = {}
-        merged_dict['slurm-args'] = default_slurm
-        merged_dict['xnat2bids-args'] = default_x2b
-
-        arglist = merged_dict
+        # Merge with default configuration
+        arglist = mergeConfigFiles(user_params, default_params)
 
     # Fetch user credentials 
     user = input('Enter Username: ')
     password = getpass('Enter Password: ')
 
-    # Fetch compiled xnat2bids and slurm parameter lists
-    x2b_param_list, slurm_param_list, bindings = compileArgumentList(arglist, user)
-
-    logging.debug("Argument Lists")
-    logging.debug("-------------------------------------")
-    logging.debug("xnat2bids: %s", x2b_param_list)
-    logging.debug("slurm: %s", slurm_param_list)
-    logging.debug("-------------------------------------")
-
-    # Insert username and password into x2b_param_list
-    x2b_param_list.insert(2, f"--user {user}")
-    x2b_param_list.insert(3, f"--pass {password}")
-
-    # Fetch latest version if not provided
-    if not ('version' in arglist['xnat2bids-args']):
-        list_of_versions = glob.glob('/gpfs/data/bnc/simgs/brownbnc/*') 
-        latest_version = max(list_of_versions, key=os.path.getctime)
-        version = latest_version.split('-')[-1].replace('.sif', '')
-
-    # Define singularity image 
-    simg=f"/gpfs/data/bnc/simgs/brownbnc/xnat-tools-{version}.sif"
-
-    # Define output for logs
-    if not ('output' in arglist['slurm-args']):
-        output = f"/users/{user}/logs/%J.txt"
-        arg = f"--output {output}"
-        slurm_param_list.append(arg)
-
-    if not (os.path.exists(os.path.dirname(output))):
-        os.mkdir(os.path.dirname(output))
-
-    # Define bids root directory
-    if not ('bids_root' in arglist['xnat2bids-args']):
-        bids_root = f"/users/{user}/bids-export/"
-        x2b_param_list.insert(1, bids_root)
-        bindings.append(bids_root)
-
-    if not (os.path.exists(os.path.dirname(bids_root))):
-        os.mkdir(os.path.dirname(bids_root))
-
-    # Compile bindings into formated string
-    bindings_str = ' '.join(f"-B {path}" for path in bindings)
-
-    # Process command string for SRUN
-    srun_cmd = shlex.split(f"srun {' '.join(slurm_param_list)} \
-        singularity exec --no-home {bindings_str} {simg} \
-        xnat2bids {' '.join(x2b_param_list)}")
-
-    logging.debug("Running xnat2bids")
-    logging.debug("-------------------------------------")
-    logging.debug("Command Input: %s", srun_cmd)
-    logging.debug("-------------------------------------")
-    
-    # Run xnat2bids
-    subprocess.run(srun_cmd)
+    # Fetch number of sessions to process
+    num_sessions = len(arglist['xnat2bids-args']['sessions'])
+
+    # Assemble parameter lists per session
+    argument_lists = []
+    for session in range(num_sessions):
```

Comment:
Why a range instead of the session object here? Why not:

```python
for session in arglist['xnat2bids-args']['sessions']
    ...
```

It seems like you're using the index below? Do you need that index?